build/src/javascript/view/columnView.js

```diff
@@ -0,0 +1,211 @@
+    <div class="card-contents-wrapper row-sort">
+      <div class="card-text-area">
+        <div class="card-title title-font">${todo.title}</div>
+        <div class="card-details">${todo.detail}</div>
+      </div>
+      <figure class="card-cross-button">
+        <img
+          src="./image/crossBtn.svg"
+          alt="cross-button-img"
+          class="cross-button-img"
+        />
+      </figure>
+    </div>
+    <div class="card-buttons-wrapper">
+      <button class="cancel-button">취소</button>
+      <button class="register-button">등록</button>
+    </div>
+  </div>`;
+    return template;
+  }, "");
+  $cards.insertAdjacentHTML("afterbegin", cardTemplate);
+  addRegisterBtnsListener($column);
+  addCardDoubleClickEvent($cards);
+  addCardsDragEvent($cards);
+}
+
+export function addPlusBtnEvent() {
+  const $columns = document.querySelectorAll(".column");
+  $columns.forEach(($column) => {
+    const $plusBtn = $column.querySelector(".plus-button");
+    const $cards = $column.querySelector(".cards");
+    $plusBtn.addEventListener("click", () => {
+      renderRegisterCard($cards);
+    });
+  });
+}
+
+function addCardsDragEvent($cards) {
+  const cardsList = $cards.querySelectorAll(".card");
+  for (const card of cardsList) {
+    addCardDragEvent(card);
+  }
+}
+
+export function addCardDragEvent(card) {
+  card.addEventListener("mousedown", (event) => {
+    const $card = event.target.closest(".card");
+    shiftX = event.pageX - $card.offsetWidth / 2;
+    shiftY = event.pageY - $card.offsetHeight / 2;
+    isDowned = true;
+    dragStart = true;
+  });
+  card.addEventListener("mousemove", () => {
+    if (isDowned) {
+      isDraggable = true;
+    }
+  });
+}
+
+export function addmouseMoveEvent() {
+  document.body.addEventListener("mousemove", handleMouseMoveEvent);
+}
+
+function addCopiedCardEvent() {
+  const $copiedCard = document.querySelector(".copied-card");
+  $copiedCard.addEventListener("mouseup", () => {
+    isDowned = false;
+    dragStart = false;
+    isDraggable = false;
+    $copiedCard.remove();
+    changeOriginalCardStyle($originalCard);
+    updateDraggedCardData();
+    $originalCard.classList.remove("original-card");
+  });
+}
+
+function handleMouseMoveEvent(event) {
+  const delayTime = 20;
+  const columnsLocation = getColumnsLocation();
+  const $card = event.target.closest(".card");
+  if (isDraggable && dragStart && $card) {
+    $originalCard = $card;
+    makeDragElement($card);
+    applyPlaceStyle($card);
+    dragStart = false;
+    addCopiedCardEvent();
+  }
+  if (isDraggable && $card) {
+    debounce(function () {
+      moveCopiedCard(event, columnsLocation, $originalCard);
+    }, delayTime)();
+  }
+}
+
+function getColumnsLocation() {
+  const columnsArr = document.querySelectorAll(".column");
+  const columnsLocation = [];
+  for (const column of columnsArr) {
+    columnsLocation.push(column.getBoundingClientRect());
+  }
+  return columnsLocation;
+}
```

Comment:
배열을 반환할 거라면, map 메서드를 사용하면 더 좋을 것 같아요.
```suggestion
  const columnsArr = document.querySelectorAll(".column");
  return columnsArr.map((column) => column.getBoundingClientRect());
}
```

Reply:
querySelectorAll로 선택한  columns를 깊은 복사를 통해 배열을 생성하고 map을 실행해서 새로운 배열을 반환하는 구조로 변경했습니다!
```javascript
const columns = document.querySelectorAll(".column");
  const columnsArr = [...columns];
  const columnsLocation = columnsArr.map((column) =>
    column.getBoundingClientRect()
  );
```

Reply:
spread operator를 사용하셨군요! 좋은 시도입니당.
다만 spread operator는 깊은 복사가 아니라 1레벨 depth만 복사하는 얕은 복사예요!
깊은 복사를 할 수 있는 메서드를 제공하는 라이브러리도 있긴 하지만,
라이브러리의 함수를 가져다 쓰기 전에 깊은 복사를 먼저 직접 구현해보시는 것도 재미있을 것 같아요 😆

배열의 map 메서드를 사용하기 위해 NodeList로 반환된 columns를 배열로 바꿔주는 과정이 필요했던 것이었군요!
제가 놓쳤었네요~~
전개 연산자(spread operator)를 사용하는 방법도 좋고, `Array.from`을 사용해서 유사배열객체나 이터러블을 배열로 바꾸는 방법도 있으니 참고해주세요 👍